pkg/network/ovn_kubernetes.go

```diff
@@ -231,6 +231,11 @@ func fillOVNKubernetesDefaults(conf, previous *operv1.NetworkSpec, hostMTU int)
 		conf.DefaultNetwork.OVNKubernetesConfig = &operv1.OVNKubernetesConfig{}
 	}
 
+	var encapOverhead uint32 = 100 // 100 byte geneve header
```

Comment:
We actually only need 58 bytes, and the rest is overkill. So you could get away with a total overhead of 58+46=104.

Well, unless the cluster is using IPv6 in which case we need another 20 bytes. See #720. We had avoided merging that because we didn't want to tighten up the MTU until we had the ability to change it after install time. Need to decide what we're doing here.

I guess one possibility would be to use an overhead of 124 (58 geneve, 20 IPv6, 46 IPsec) when using IPsec for now, and then once we have the ability to change it post-install, we can fix it to only use the 20 bytes for IPv6 if they're actually needed.

Reply:
I feel that is a bit out of the scope of this PR. I want to only decrease the MTU by the IPsec overhead. I don't know the background to those changes. Maybe I can do something like this somewhere just to make it clearer to the reader:

```
encapOverhead += geneveOverhead
```

Reply:
well, note at least that you are currently reserving 146 bytes for overhead, but per the math above, even if you want to unconditionally leave room for IPv6, you only need 124 bytes. So you could just say

```
	var encapOverhead uint32 = 100
	if conf.DefaultNetwork.OVNKubernetesConfig.IPsecConfig != nil {
		encapOverhead = 124
	}
```

Reply:
The IPsec overhead is 46 bytes on top of Geneve so I am just adding another 46 onto the (already specified) overhead. Basically, I want to call out all the overheads separately and add them on as needed. I will rework this (it will be a separate PR - https://github.com/openshift/cluster-network-operator/pull/928 ) and you can have a look and comment there if you would like me to change

> Can you also grab [danwinship@6f1e7b5](https://github.com/danwinship/cluster-network-operator/commit/6f1e7b5e) on to this branch?

I'm splitting this PR up so that the BZs line up correctly but I will add it to another PR if you would like